google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddRemarketingAction.java

```diff
@@ -0,0 +1,152 @@
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddRemarketingAction().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+
+    // Creates a remarketing action with the specified name.
+    RemarketingAction remarketingAction =
+        RemarketingAction.newBuilder()
+            .setName(StringValue.of("Remarketing action #" + System.currentTimeMillis()))
+            .build();
+
+    // Creates a remarketing action operation.
+    RemarketingActionOperation operation =
+        RemarketingActionOperation.newBuilder().setCreate(remarketingAction).build();
+
+    // Issues a mutate request to add the remarketing action and prints out some information.
+    String remarketingActionResourceName;
+    try (RemarketingActionServiceClient conversionActionServiceClient =
+        googleAdsClient.getLatestVersion().createRemarketingActionServiceClient()) {
+      MutateRemarketingActionsResponse response =
+          conversionActionServiceClient.mutateRemarketingActions(
+              Long.toString(customerId), Collections.singletonList(operation));
+      remarketingActionResourceName = response.getResults(0).getResourceName();
+      System.out.printf(
+          "Added remarketing action with resource name '%s'.%n", remarketingActionResourceName);
+    }
+
+    // Creates a query that retrieves the previously created remarketing action with its generated
```

Comment:
This may be cleaner and easier to maintain as a separate getRemarketingAction method.

Reply:
Discussed offline. For consistency across all libraries, agreed to leave this as-is.